express-auth/auth.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  type NextFunction,
+  type Request,
+  type Response,
+  Router,
+} from "express";
+import createExpressAuth, {
+  AuthRequest,
+  CallbackRequest,
+  ExpressAuthSession,
+} from "@edgedb/auth-express";
+import { client } from "./db";
+
+export const auth = createExpressAuth(client, {
+  baseUrl: "http://localhost:3333",
+});
+
+export const requireAuth = async (
+  req: AuthRequest,
+  res: Response<any, { session?: ExpressAuthSession }>,
+  next: NextFunction
+) => {
+  if (!(await req.session?.isSignedIn())) {
+    res.redirect("/signin");
+  } else {
+    next();
+  }
+};
+
+/************ 
+ * Sign Out *
+ ************/
+
+export const signoutRoute = Router().get(
+  "/",
+  auth.signout,
+  async (_: AuthRequest, res: Response) => {
+    res.redirect("/");
+  }
+);
+
+/*************** 
+ * Built-in UI *
+ ***************/
+
+const builtinCallback = async (req: CallbackRequest, res: Response) => {
+  if (req.isSignUp) {
+    return res.redirect("/onboarding");
+  }
+
+  res.redirect("/");
+};
+
+export const factoriedBuiltinUIRouter = auth.createBuiltinRouter({
+  callback: [builtinCallback],
+});
+
+export const builtinUIRouter = Router()
```

Comment:
Actually, I wonder if I should demonstrate both, since the manual router needs to mounted to a route since it doesn't define a route itself. Maybe I can make two apps, one that uses the manual router and has a custom UI, and one that uses the built-in UI with the factory-style router?

That would address both of these "dead-code" suggestions. WDYT?

Reply:
Yes! I like the idea of helping people connect those dots.

Reply:
⚡ @raddevon https://github.com/edgedb/edgedb-examples/pull/116/files/1e78110c50e05ff38a17f3f0d7c1e26313136281..fec57be36a6e1699186a12e4b2631311dfccea96